lib/src/http.dart

```diff
@@ -38,14 +40,24 @@ final _sessionId = createUuid();
 /// user-friendly error messages.
 class _PubHttpClient extends http.BaseClient {
   final _requestStopwatches = <http.BaseRequest, Stopwatch>{};
+  SystemCache get cache => _cache ??= SystemCache();
```

Comment:
Why not pass in the `cache` from the `PubCommand`? This will duplicate the caching and may break a (future) assumption that the cache is a singleton.

Reply:
I don't think this is possible. The _PubHttpClient is constructed through _pubClient variable, which is directly used in the RetryClient constructor, which is used in the _ThrowingClient constructor, which is only used in the httpClient = ThrottleClient which is a public field in the http.dart that is used everywhere...
So no way to pass it around I guess.